userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -1402,6 +1402,7 @@ static int32_t set_default_settings(struct bpf_engine *handle)
 	settings.do_dynamic_snaplen = false;
 	settings.dropping_mode = false;
 	settings.is_dropping = false;
```

Comment:
Not a comment related to this PR, but perhaps in a different PR we should cleanup `settings.is_dropping` and `settings.dropping_mode` (which @FedeDP it's still in your queue right?) and maybe also come up with more concrete names to convey the intend better (`UF_NEVER_SAMPLE`, `UF_ALWAYS_SAMPLE` or similar). Not a secret that I am not a big fan of any of the existing logic related to this. 

Maybe it could be made clearer that this is "random sampling" for non Falco use cases such as clients doing system diagnostic. That way it would relate better to `sampling_ratio` ... and since we now have a clean interface for either tracing or not tracing a particular syscall, there should really be only sampling or no sampling when a syscall is being traced (if you opt-into this behavior in the first place).

Reply:
Yep, i am neither...but since that logic has been in place for ages, i am not sure we should kill it.
But i am definitely looking forward for the moment we might be able to :fire: it :laughing: 

@gnosek :P

Reply:
:) It's ok to keep a sampling mechanism, just revamp it a bit as suggested above especially wrt naming and the fact we now have the new "drop an uninteresting syscall" which didn't exist when that logic was invented very long time ago.